src/Result/JunitFormatter.php

```diff
@@ -117,12 +123,16 @@ public function format(
 
         if ($unusedIgnores !== [] && $configuration->shouldReportUnmatchedIgnoredErrors()) {
             $hasError = true;
-            $xml .= $this->createUnusedIgnoresTestSuite($unusedIgnores);
+            $this->createUnusedIgnoresTestSuite($unusedIgnores);
         }
 
-        $xml .= '</testsuites>';
+        $xmlString = $this->document->saveXML(null, LIBXML_NOEMPTYTAG);
 
-        $this->printer->print($xml);
+        if ($xmlString === false) {
+            $xmlString = '';
```

Comment:
Isnt this hiding some failure? I think this should be some exception.

Reply:
Maybe you need something like [this](https://github.com/lightools/xml/blob/master/src/Xml/XmlLoader.php#L48) to get proper exception message.

Not sure tho...